lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
Ok.. i see what you mean, so should we remove this clause bellow?
```elixir
  defp random_integer(limit, limit) when is_integer(limit),
    do: limit
```
and also get rid of the optimizations

```elixir
  def take_random(_enumerable, 0),
    do: []
  def take_random(first..first, count) when is_integer(count) and count >= 1,
    do: [first]
```